lib/runa/client.rb

```diff
@@ -32,10 +30,14 @@ def initialize(options = {})
       end
     end
 
-    def request(method, path, payload = {})
+    def request(method, path, payload = {}, key = '')
```

Comment:
no need to set a default value here, in fact it could be dangerous. If we wanted to make this safer, we should check this value for POST and throw an error if it's nil or empty. You can do that easily in Ruby with `.blank?`

Reply:
```ruby
if method.eql? :post
    req.headers['X-Idempotency-Key'] = key unless key.blank?
end```

will this inherently throw an error or do I need to explicitly throw one?

Reply:
It won't throw an error no, that as pseudocode is "unless key is blank, assign the value of key to req.headers['X-Idempotency-Key'], otherwise do nothing"

Reply:
If you want to explicitly raise an error in Ruby would do something like:
```ruby
raise StandardError('Required idempotency key missing') if key.blank?
```


I would think about creating a custom error class for this case as well, but we can cover that separately.